backend/src/Entity/DocumentCategory.php

```diff
@@ -28,7 +43,7 @@ public function getName(): ?string
 
     public function __toString(): string
     {
-        return $this->getName();
+        return $this->parent ? $this->parent . ' > ' . $this->name : $this->name;
```

Comment:
i believe you need to call the getName() method on the parent:
return $this->parent ? $this->parent->getName() . ' > ' . $this->name : $this->name;